pkg/artifacts/signable.go

```diff
@@ -366,13 +369,27 @@ func isStructuredResult(res objects.Result, categoryMarker string) (bool, error)
 }
 
 func checkDigest(dig string) error {
-	prefix := digest.Canonical.String() + ":"
-	if !strings.HasPrefix(dig, prefix) {
-		return fmt.Errorf("unsupported digest algorithm: %s", dig)
+	// 93c56eeba9ec70f74c9bfd297d9516642d366cb5
+	parts := strings.Split(dig, ":")
+	if len(parts) != 2 {
+		return fmt.Errorf("digest string %s, not in the format of <algorithm>:<digest>", dig)
 	}
-	hex := strings.TrimPrefix(dig, prefix)
-	if err := digest.Canonical.Validate(hex); err != nil {
-		return err
+	algo_string := strings.ToLower(strings.TrimSpace(parts[0]))
+	algo := digest.Algorithm(algo_string)
+	hex := strings.TrimSpace(parts[1])
+
+	switch {
+	case algo.Available():
+		if err := algo.Validate(hex); err != nil {
+			return err
+		}
+	case algo_string == "sha1":
+		if !regexp.MustCompile(`^[a-f0-9]{40}$`).MatchString(hex) {
```

Comment:
Maybe `regexp.MustCompile` should be called at the package level? Minor performance boost.